tools/genSettings.py

```diff
@@ -4,6 +4,90 @@
 import sys
 import json
 
+def writeHelper(f, obj, pre, indent):
+    ids = "    " * indent
+    for t in obj:
+        name = t["name"]
+        key = name.replace(" ", "").replace("/", "").replace("+", "p")
+        typename = t["type"]
+        if typename == "Object":
+            f.write(f"{ids}struct {key} {{\n")
+            f.write(f"{ids}    QString pre;\n")
+            f.write(f"{ids}    {key}(QString p) : pre(p) {{}}\n")
+            writeHelper(f, t["sub"], "pre, ", indent + 1)
+            f.write(f"{ids}}};\n")
+            f.write(f"{ids}inline {key} get{key}(QString key) {{ return {key}(QStringList {{{pre} {json.dumps(name)}, key}}.join('/')); }}\n")
+            f.write(f"{ids}inline void remove{key}(QString key) {{ SettingsManager::remove(SettingsManager::keyStartsWith(QStringList {{{pre} {json.dumps(name)}, key}}.join('/'))); }}\n")
+            f.write(f"{ids}inline QStringList query{key}() {{ return SettingsManager::itemUnder(QStringList {{{pre} {json.dumps(name)}}}.join('/') + '/'); }}\n")
```

Comment:
```suggestion
            f.write(f"{ids}inline QStringList query{key}() {{ return SettingsManager::itemUnder(QStringList {{{pre} {json.dumps(name)}}}.join('/') + '/'); }}\n")
```

I think this can also be `get`, why `query`?

Reply:
queryXXX return all entrys and getXXX return certain data

Reply:
Why not `getSnippet` and `getSnippets`?